lib/segment/src/vector_storage/chunked_utils.rs

```diff
@@ -48,6 +48,16 @@ pub fn read_mmaps<T: Sized>(directory: &Path) -> OperationResult<Vec<MmapChunk<T
             ))
         })?;
         let mmap = open_write_mmap(&mmap_file)?;
+        // If unix, lock the memory
+        #[cfg(unix)]
+        if mlock {
+            mmap.lock()?;
```

Comment:
I believe I've seen this fail in some cases, for example when low in memory. I therefore think we should return a warning here rather than failing.